test-requirements.txt

```diff
@@ -0,0 +1,3 @@
+pytest
```

Comment:
Do these requirements overlap with dev-requirements? We might want to think about restructuring our requirements files (concerned about redundancy; easy to make updates in one and not another, e.g. like the recent pytest version issue we ran into).  I'm ok with this for now, but would like to revisit soon (maybe before we adapt for another project).

Reply:
hmm, that's a good point. I'm not sure what to call/how to think of those shared requirements, but it's definitely true that there's duplication we could avoid.

Reply:
I've seen directories and includes used before, something like the structure described here: https://stackoverflow.com/questions/17803829/how-to-customize-a-requirements-txt-for-multiple-environments

This actually could simplify things - you ought to be able to install just one set of things and get everything you need.  We could still keep a top-level requirements.txt that includes the appropriate requirements/whatever.txt if that helps with deploy and travis-ci configuration.

We might want to shift to something like this now that we have a need for test requirements that exclude dev requirements. Not sure it should be in scope for this PR.

@bwhicks what do you think?

Reply:
@rlskoeser I've been having similar thoughts, especially now that our requirements are growing and getting a bit more complex. I wish python had something closer to npm in terms of fine-grained control.

I agree we should do it and agree that it's out of scope for this PR.